src/components/activity-groups-list/activity-groups-list.tsx

```diff
@@ -30,22 +41,23 @@ export const ActivityGroupsList: FC<Props> = ({
   const fakeRefreshControlProps = useFakeRefreshControlProps();
 
   const sections = useMemo(() => {
-    const result = [];
+    const result: ListItem[] = [];
     let prevActivityDate = new Date(-1);
 
     for (const activityGroup of activityGroups) {
       const firstActivity = activityGroup[0] ?? emptyActivity;
       const date = new Date(firstActivity.timestamp);
+      const lastActivityGroup = result[result.length - 1];
 
-      if (isTheSameDay(date, prevActivityDate)) {
-        result[result.length - 1].data.push(activityGroup);
+      if (isTheSameDay(date, prevActivityDate) && Array.isArray(lastActivityGroup)) {
+        lastActivityGroup.concat(activityGroup);
```

Comment:
Sorry, `concat` function doesn't mutate an array instance
```suggestion
        result[result.length - 1] = lastActivityGroup.concat(activityGroup);
```